src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1470,3 +1470,11 @@ def list_consumption_locations():
     client = web_client_factory()
     regions = client.list_geo_regions(sku='Dynamic')
     return [{'name': x.name.lower().replace(" ", "")} for x in regions]
+
+def list_locations(sku):
+    client = web_client_factory()
```

Comment:
Create the client close to where it is needed. The `_get_sku_name` may throw an exception and interrupt the flow.